### PR TITLE
docs(docs): added dynamic component docs

### PR DESCRIPTION
**Pull request by jeffdowdle**
**Issue**:

### What I did

- 
- 

### How to test

- 
- 

#### For all PR's

- [ ] I've added relevant changes to the project Readme if needed.
- [ ] I've updated the documentation site as needed.
- [ ] I have added unit tests to cover my changes (if not applicable, please state why in a comment)

#### For new components only

- [ ] I have added a story covering all variants
- [ ] I have checked a11y tab in storybook passes
- [ ] Any events are emitted on the event bus

### CODE REVIEW

**Comment by lambry** [comment]
@jeffdowdle Do we want to mention the mapping utils that are available i.e. `getField`, `getBody`, `getImageFromField`, `getLinkFromField`, `getBodyFromField`

File: docs/content/framework/3.guides/5.dynamic-components.md [hunk 1]
```diff
@@ -5,8 +5,121 @@ layout: page
...
+<template>
+  <div>
+    {{someProp2}}
+  </div>
+</template>
+```
+
+## The list of includes
+
+Each dynamic component should specify an array of 'includes' that it needs in order to function properly.
```
Comment (dylankelly) [inline docs/content/framework/3.guides/5.dynamic-components.md:107]: @jeffdowdle probably need something in the concept page to talk to this, probably with a link to this https://www.drupal.org/docs/core-modules-and-themes/core-modules/jsonapi-module/includes

File: docs/content/framework/3.guides/5.dynamic-components.md [hunk 2]
```diff
@@ -5,8 +5,121 @@ layout: page
...
+
+```ts
+const includes = [
+  'field_landing_page_component.field_paragraph_accordion'
+]
+```
+
+## The list of content types
+
+Each dynamic component should specify an array of 'content types' that the dynamic component can appear on.
```
Comment (dylankelly) [inline docs/content/framework/3.guides/5.dynamic-components.md:117]: Each dynamic component should specify an array of 'content types' that the dynamic component can appear on. This should be the value of the `type` property in the Drupal JSON API response.

**Comment by jeffdowdle** [comment]
> @jeffdowdle Do we want to mention the mapping utils that are available i.e. `getField`, `getBody`, `getImageFromField`, `getLinkFromField`, `getBodyFromField`

I think this kind of 'reference' information is better suited to the readme of the ripple-tide-api package

**Review by lambry — APPROVED** [review]

**Review by dylankelly — APPROVED** [review]

**Pull request merged**